typescript/src/bitcoin.ts

```diff
@@ -359,3 +361,40 @@ export function createKeyRing(
 export function computeHash160(text: string): string {
   return hash160.digest(Buffer.from(text, "hex")).toString("hex")
 }
+
+/**
+ * Computes the SHA256 for the given text.
+ * @param text - Text the SHA256 is computed for.
+ * @returns Hashed hex string
+ */
+export function computeSha256(text: string): string {
+  return sha256.digest(Buffer.from(text, "hex")).toString("hex")
+}
+
+/**
+ * Encodes plain text to base58 encoded data
+ * @param text - Text that will be encoded
+ * @returns Encoded base58 text
+ */
+export function encodeBase58(text: string): string {
+  return base58.encode(Buffer.from(text, "hex"))
+}
+
+/**
+ * Decodes base58 encoded data to plain text
+ * @param text - Base58 encoded text that will be decoded
+ * @returns Decoded text
+ */
+export function decodeBase58(text: string): string {
+  return base58.decode(text).toString("hex")
+}
+
+/**
+ * Encodes public key hash into a P2PKH address
+ * @param publicKeyHash - public key hash that will be encoded
+ * @returns P2PKH address encoded from the given public key hash
+ */
+export function generateP2PKHAddress(publicKeyHash: string) {
```

Comment:
The logic is wrong as the checksum is used in a wrong way. Only 4 first bytes should be taken. Additionally, the `00` is not universal and is network-specific. It will be different for mainnet and testnet addresses. See https://www.oreilly.com/library/view/mastering-bitcoin/9781491902639/ch04.html for reference. 

However, you don't have to do the encoding manually as we can use:
```
bcoin.Address.fromPubkeyhash(Buffer.from(publicKeyHash, "hex"))`
```
to construct a bcoin's `Address` object and then get the string version using `toString(network)`. The `publicKeyHash` must be a HEX string, non-prefixed with `0x`. Please note that in the parameter documentation.
The `network` argument accepted by `toString` implies that this function should accept the `network` parameter as well, likewise the `calculateDepositAddress` function. 

I'd also rename `generateP2PKHAddress` to `encodeBitcoinAddress`. We don't "generate" anything here.

Reply:
One important note, this function should also accept `witness` parameter and use `fromWitnessPubkeyhash` if `witness` is true and `fromPubkeyhash` otherwise. The address is encoded differently for SegWit hence that parameter.

Reply:
Good point, `encodeToBitcoinAddress` will be a good option!